backend/test_observer/data_access/models.py

```diff
@@ -225,6 +235,11 @@ class TestExecution(Base):
     status: Mapped[TestExecutionStatus] = mapped_column(
         default=TestExecutionStatus.NOT_STARTED
     )
+    review_status: Mapped[list[TestExecutionReviewStatus]] = mapped_column(
+        ARRAY(Enum(TestExecutionReviewStatus)),
+        default=[TestExecutionReviewStatus.UNDECIDED],
+    )
```

Comment:
It's because we established that multiple reasons for approving might apply, even per environment.

Reply:
Ooh, that's a good point. But from a design perspective, I'm not sure this is clear or clean. It feels like we want to have comments/reasons/approval per TestResult but at the level of TestExecution somehow. I'd prefer if either we go full on to the level of TestResult or just stay at TestExecution or even just Artefacts with comments. I'm trying to think of some conceptual way to simplify this, as it seems over complicated right now in my head (both code and simply understanding from user perspective)

Reply:
We were also discussing this before with @mz2, however the other alternative was to have approval on the level of different Test Cases, which might add more complexity to the whole workflow, especially if more than a few test cases fail because of the same error, you would need to go through all of them and mark them as `APPROVED`, so we decided to keep some balance and have approval on the level of Test Execution, similarly to what we had in Trello.

Reply:
I'm thinking of complexity as how much explanation needs to be given to a user or developer prior to working on the feature at hand. By that definition I feel that if we take the approach of a list of approval states, then we have a lot of explanation to do for both developers and users. On the other hand the approaches below require much less explanation if any:
1. Approval per test result. If test result model had a review state it's sort of clear. The UI is also clear as we will have review button next to each test result that failed.
2. Approval per test execution but with only one approval state. Any explanation is in comments. Yes we can do less automation but this is very flexible and simple (automation on the list approach is hard too)
3. Approval per artefact but with only one approval state. Any explanation is in comments. From my understanding, this is what we currently have on trello. Given that we're replacing trello we can start with same functionality then improve once we have a better design and purpose

Reply:
Following a short discussion with @mz2 on MM, there are a few reasons why we should go with the current workflow:
* Having a list of approval reasons to pick will let us improve our process overall. For example, we can easily find devices that lead to failing test cases very often, and use this knowledge to make decisions on what to do with them. Another type of analysis we might be able to do is to find tests that fail often and focus on improving them.
* Since we are implementing a new workflow, it might be better to focus on making it good now, instead of reworking it later once again.

Reply:
Looking at both these points, I then see that option 1 is better. It gives us more control to analyze and report exactly what happened and what we approved. We can have an enum of approval reasons that we select only one option from (not a list)

Reply:
The option (1) would indeed be the most accurate solution, it was something I was going to request initially (we also discussed it last week), but it leads to a lot of UX friction: if there are 100 failed tests because the hardware is faulty, for example, you have a 100 buttons to press. I was trying to propose this current level of granularity as a compromise between the user friction and the reporting standard.

Reply:
> we will still have to answer whether there is approval per test execution too?

There should IMO downstream of this per artifact review choice be an option to "ignore" or "dismiss" a certain specific test result and to make that dismissal persistent, so that a test when executed on a specific device would not prevent signing off automatically even if it fails. This is going to be helpful for use of Test Observer with a) prerelease use cases (known issues are acceptable for an edge kernel, for example) and b) avoiding manual busywork when there are known-to-be-problematic test cases + test devices combos in the pool.

When implemented, that will need to come with similarly an explicit set of options for explaining why (and those may be the same review states we are introducing here).

Basically, I think the per execution level detail will _also_ be needed, just not yet and it will have a different supporting use.